site/app/controllers/course/CourseMaterialsController.php

```diff
@@ -263,6 +263,52 @@ public function modifyCourseMaterialsFileTimeStamp($filenames, $newdatatime) {
 
         return $this->core->getOutput()->renderResultMessage("Time successfully set.", true);
     }
+    
+    /**
+     * @Route("/{_semester}/{_course}/course_materials/edit", methods={"POST"})
+     * @AccessControl(role="INSTRUCTOR")
+     */
+    public function ajaxEditCourseMaterialsFiles() {
+        $sections = null;
+        if (isset($_POST['sections'])) {
+            $sections = $_POST['sections'];
+        }
+        
+        $hide_from_students = null;
+        if (isset($_POST['hide_from_students'])) {
+            $hide_from_students = $_POST['hide_from_students'];
+        }
+        
+        if (empty($sections) && !is_null($sections)) {
+            $sections = [];
+        }
+        
+        $requested_path = "";
+        if (isset($_POST['requested_path'])) {
+            $requested_path = $_POST['requested_path'];
+        }
+        
+        $sections_exploded = $sections;
+        
+        if (!(is_null($sections)) && !empty($sections)) {
+            $sections_exploded = [];
+            $sections_exploded = explode(",", $sections);
+        }
+        
+        $release_time = "";
+        if (isset($_POST['release_time'])) {
+            $release_time = $_POST['release_time'];
+        }
+        
+        $fp = $this->core->getConfig()->getCoursePath() . '/uploads/course_materials_file_data.json';
```

Comment:
`$fp = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), 'uploads', 'course_materials_file_data.json');`